tests/components/logger/test_init.py

```diff
@@ -25,6 +25,54 @@ def restore_logging_class():
     logging.setLoggerClass(klass)
 
 
+async def test_log_filtering(hass):
```

Comment:
We do have a `caplog` fixture that could be used to provide a more end-to-end test.

Reply:
Good to know. I've updated the tests. I also added a test case to ensure filters are scoped to their respective services.